test/unit/timeout.test.ts

```diff
@@ -115,3 +123,202 @@ describe('Timeout', function () {
     });
   });
 });
+
+describe('TimeoutContext', function () {
+  describe('TimeoutContext.create', function () {
+    describe('when timeoutMS is a number', function () {
+      it('returns a CSOTTimeoutContext instance', function () {
+        const ctx = TimeoutContext.create({
+          timeoutMS: 0,
+          serverSelectionTimeoutMS: 0,
+          waitQueueTimeoutMS: 0
+        });
+
+        expect(ctx).to.be.instanceOf(CSOTTimeoutContext);
+      });
+    });
+
+    describe('when timeoutMS is undefined', function () {
+      it('returns a LegacyTimeoutContext instance', function () {
+        const ctx = TimeoutContext.create({
+          serverSelectionTimeoutMS: 0,
+          waitQueueTimeoutMS: 0
+        });
+
+        expect(ctx).to.be.instanceOf(LegacyTimeoutContext);
+      });
+    });
+  });
+
+  describe('CSOTTimeoutContext', function () {
+    let ctx: CSOTTimeoutContext;
+
+    afterEach(() => {
+      // @ts-expect-error private property
+      ctx.overallTimeout?.clear();
+    });
```

Comment:
```suggestion
```

as far as I can tell, this property no longer exists on the CSOTTimeoutContext.  right?